tests/unit/test_utils.py

```diff
@@ -465,23 +465,29 @@ def test_auto_decode_pep263_headers(self):
         assert auto_decode(latin1_req.encode('latin1')) == latin1_req
 
 
-class TestBuildDirectory(object):
+class TestTempDirectory(object):
```

Comment:
We should include additional tests. For example, that the directory is cleaned up properly after use. Especially if there are files with funky permissions in there (I'd particularly like to see a test here that readonly files on Windows don't break the cleanup, as git uses these - the code in ```pip.util.rmtree``` should cover this, but it's something we should explicitly confirm for ```TempDir```).